src/domain/collaboration/opportunity/opportunity.resolver.mutations.ts

```diff
@@ -169,4 +172,46 @@ export class OpportunityResolverMutations {
       agentInfo
     );
   }
+
+  @UseGuards(GraphqlGuard)
+  @Mutation(() => IUser, {
+    description: 'Assigns a User as an Opportunity Admin.',
+  })
+  @Profiling.api
+  async assignUserAsOpportunityAdmin(
+    @CurrentUser() agentInfo: AgentInfo,
+    @Args('membershipData') membershipData: AssignOpportunityAdminInput
+  ): Promise<IUser> {
+    const opportunity = await this.opportunityService.getOpportunityOrFail(
```

Comment:
I know this is done consistently everywhere, but it's all copy-pasted (but one line). My suggestion is to move most of this logic in the xxxService (in this case opportunityService) and have the common logic in a method.